hr_employee_lastnames/models/hr_employee.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+from odoo.addons.hr_employee_firstname.models.hr_employee import UPDATE_PARTNER_FIELDS
+
+UPDATE_PARTNER_FIELDS += ['lastname2']
+
+
+class HrEmployee(models.Model):
+    _inherit = 'hr.employee'
+
+    lastname2 = fields.Char("Second last name")
+
+    @api.model
+    def _get_name_lastnames(self, lastname, firstname, lastname2=None):
+        return self.env['res.partner']._get_computed_name(
+            lastname, firstname, lastname2)
+
+    def _build_name_in_create(self, vals):
+        if any([field in vals for field in ['firstname', 'lastname', 'lastname2']]):
+            vals['name'] = self._get_name_lastnames(
+                vals.get('lastname'), vals.get('firstname'), vals.get('lastname2'))
+        elif vals.get('name'):
+            vals['firstname'] = self.split_name(vals['name'])['firstname']
+            vals['lastname'] = self.split_name(vals['name'])['lastname']
+            vals['lastname2'] = self.split_name(vals['name'])['lastname2']
+        else:
+            raise UserError(_('No name set.'))
+        return vals
+
+    def _build_name_in_write(self, vals):
+        if any([field in vals for field in ['firstname', 'lastname', 'lastname2']]):
+            if 'lastname' in vals:
+                lastname = vals.get('lastname')
```

Comment:
Don't use `.get` if you are sure that this key exists and the `if key in vals` ensure that it exists

```suggestion
                lastname = vals['lastname']
```

Reply:
SyntaxError: Generator expression must be parenthesized if not sole argument
because python2.7

Reply:
sorry, wrong place

Reply:
done